### ISSUE TITLE
Impossible to attach image as an attachment to a test case

### ISSUE THREAD

**Issue by anto-wahanda** [webdriverio/wdio-allure-reporter#98]
# Description of the issue

When trying to add an image as attachment to the report for a test case using `reporter.createAttachment()`, an attachment is added to the xml report, but 

- the image generated and saved in the allure results folder cannot be open
- the attachment does not have a specified size, hence the report cannot be generated

# Steps to reproduce

1. Clone the repo with a prepared example: https://github.com/anto-wahanda/wdio-allure-attachment-issue
2. `npm run`, `npm test` and `npm run allure` (if you want to attempt generating the report)

# Actual result

Generated PNG cannot be open, attachment in the xml suite result file does not have a size, report cannot be generated.
(NOTE: the attachment generated when invoking `browser.screenshot()` is correctly attached to the xml result file under its own step)

# Expected result

PNG file should be saved correctly (i.e. you should be able to open it), generated report should show the attachment and image should be visible within the report.

**Comment by BorisOsipov** [comment]
@anto-wahanda thank you for comprehensive bug report.

Yes it is a wdio-allure-reporter issue

Here we need to pass `Buffer.from(content, type)` rather just `content`
https://github.com/webdriverio/wdio-allure-reporter/blob/6aedcd4459f5ac0d092cc197681b3d7b53520d01/lib/reporter.js#L197

**Comment by anto-wahanda** [comment]
@BorisOsipov I think it makes sense, although... what would type be in that context? If it's the type passed in when calling `createAttachment()` that would be invalid. Should it always be `'base64'`?

Also, should createAttachment take a base64 encoded string then?

**Comment by BorisOsipov** [comment]
@anto-wahanda good questions. 
>what would type be in that context?
We need to pass type. it will be requered field.
Now I don't have time to go deeper and answer.

>Should it always be 'base64'?
Maybe. I don't know are there any problem with reading base64 from unknown source.
I can advice to look how allure recognize mime type 
https://github.com/allure-framework/allure-js-commons/blob/master/util.js

Also, should createAttachment take a base64 encoded string then?
No. If we make type required.

**Comment by anto-wahanda** [comment]
I am confident I have a fix, which is basically `allure.addAttachment(name, Buffer.from(content), type)`.

I want to add a test, but I'm struggling with cheerio API. After doing the following, I want to at least verify that the attachment has a size set, but I haven't been able to figure out how.

```
const pngAttachment = getResultFiles('png')
expect(pngAttachment, 'PNG attachment file was not generated').to.have.lengthOf(1)
expect(result('test-case attachment[title="PNG file attachment"]'),
         'PNG attachment file was not added to general xml report').to.have.lengthOf(1)
```

**Comment by christian-bromann** [comment]
Merged and released as v0.3.3
